content/en/tracing/guide/tutorial-enable-java-aws-ecs-fargate.md

```diff
@@ -0,0 +1,434 @@
+   
+    For this tutorial, the `/terraform/Fargate/Instrumented/main.tf` file already has these environment variables defined for the notes and calendar applications, for example for `notes`:
+
+    ```yaml
+    ...
+    
+       name : "notes",
+       image : "${module.settings.aws_ecr_repository}:notes",
+       essential : true,
+       portMappings : [
+         {
+           containerPort : 8080,
+           hostPort : 8080
+         }
+       ],
+       memory : 512,
+       cpu : 256,
+       environment : [
+         {
+           name : "CALENDAR_HOST",
+           value : "calendar.apmlocaljava"
+         },
+         {
+           name : "DD_SERVICE",
+           value : "notes"
+         },
+         {
+           name : "DD_ENV",
+           value : "dev"
+         },
+         {
+           name : "DD_VERSION",
+           value : "0.0.1"
+         }
+       ],
+       dockerLabels : {
+         "com.datadoghq.tags.service" : "notes",
+         "com.datadoghq.tags.env" : "dev",
+         "com.datadoghq.tags.version" : "0.0.1"
+       },
+       ...
+    ```
+    And for `calendar`:
+
+    ```yaml
+     ...
+        name : "calendar",
+        image : "${module.settings.aws_ecr_repository}:calendar",
+        essential : true,
+        environment : [
+          {
+            name : "DD_SERVICE",
+            value : "calendar"
+          },
+          {
+            name : "DD_ENV",
+            value : "dev"
+          },
+          {
+            name : "DD_VERSION",
+            value : "0.0.1"
+          }
+       ],
+       dockerLabels : {
+         "com.datadoghq.tags.service" : "calendar",
+         "com.datadoghq.tags.env" : "dev",
+         "com.datadoghq.tags.version" : "0.0.1"
+       },
+      ...
+     ```
+
+    You can also see that Docker labels for the same Universal Service Tags `service`, `env`, and `version` values are set. This allows you to also get Docker metrics once your application is running.
+
+### Tracing configuration
+
+The Java tracing library takes advantage of Java’s built-in agent and monitoring support. The flag `-javaagent:../dd-java-agent.jar` in the Dockerfile tells the JVM where to find the Java tracing library so it can run as a Java Agent. Learn more about Java Agents at [https://www.baeldung.com/java-instrumentation][7].
+
+The `dd.trace.sample.rate` flag sets the sample rate for this application. The ENTRYPOINT command in the Dockerfile set its value to `1`, which means that 100% of all requests to the services are sent to the Datadog backend for analysis and display. For a low-volume test application, this is fine. Do not do this in production or in any high-volume environment, because this results in a very large volume of data. Instead, sample some of your requests. Pick a value between 0 and 1. For example, `-Ddd.trace.sample.rate=0.1` sends traces for 10% of your requests to Datadog. Read more about [tracing configuration settings][14] and [sampling mechanisms][15].
```

Comment:
```suggestion
The `dd.trace.sample.rate` flag sets the sample rate for this application. The ENTRYPOINT command in the Dockerfile sets its value to `1`, meaning that 100% of all service requests are sent to the Datadog backend for analysis and display. For a low-volume test application, this is fine. Do not do this in production or in any high-volume environment, because this results in a very large volume of data. Instead, sample some of your requests. Pick a value between 0 and 1. For example, `-Ddd.trace.sample.rate=0.1` sends traces for 10% of your requests to Datadog. Read more about [tracing configuration settings][14] and [sampling mechanisms][15].
```